ta/pkcs11/src/persistent_token.c

```diff
@@ -538,6 +539,40 @@ void release_persistent_object_attributes(struct pkcs11_object *obj)
 	obj->attributes = NULL;
 }
 
+enum pkcs11_rc update_persistent_object_attributes(struct pkcs11_object *obj)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	TEE_ObjectHandle hdl = TEE_HANDLE_NULL;
+	uint32_t tee_obj_flags = TEE_DATA_FLAG_ACCESS_WRITE;
+	size_t size = 0;
+
+	if (!obj->attributes)
+		assert(!obj->attributes);
+
+	size = sizeof(struct obj_attrs) + obj->attributes->attrs_size;
+
+	res = TEE_OpenPersistentObject(TEE_STORAGE_PRIVATE,
+				       obj->uuid, sizeof(*obj->uuid),
+				       tee_obj_flags, &hdl);
+	if (res) {
+		EMSG("OpenPersistent failed %#"PRIx32, res);
+		return tee2pkcs_error(res);
+	}
+
+	res = TEE_SeekObjectData(hdl, 0, TEE_DATA_SEEK_SET);
+	if (res)
+		goto out;
+
+	res = TEE_WriteObjectData(hdl, obj->attributes, size);
+	if (res)
+		goto out;
+
+	res = TEE_TruncateObjectData(hdl, size);
+
+out:
+	TEE_CloseObject(hdl);
+	return tee2pkcs_error(res);
+}
```

Comment:
add an empty line below

Reply:
you forgot this one :)

Reply:
Sorry again. I misunderstood that you wanted a blank line between 573 and 574 and added that. I am removing that now and adding one after 575.